x/bank/keeper/view.go

```diff
@@ -68,6 +71,31 @@ func (k BaseViewKeeper) GetAllBalances(ctx sdk.Context, addr sdk.AccAddress) sdk
 	return balances.Sort()
 }
 
+// GetAllBalancesNew returns all the account balances for the given account address.
+func (k BaseViewKeeper) QueryAllBalances(ctx sdk.Context, addr sdk.AccAddress,
+	pageReq *query.PageRequest) (sdk.Coins, *query.PageResponse, error) {
+	balances := sdk.NewCoins()
+	store := ctx.KVStore(k.storeKey)
+	balancesStore := prefix.NewStore(store, types.BalancesPrefix)
+	accountStore := prefix.NewStore(balancesStore, addr.Bytes())
+
+	res, err := query.Paginate(accountStore, pageReq, func(key []byte, value []byte) error {
+		var result sdk.Coin
+		err := k.cdc.UnmarshalBinaryBare(value, &result)
+		if err != nil {
+			return err
+		}
+		balances = append(balances, result)
+		return nil
+	})
+
+	if err != nil {
+		return sdk.NewCoins(), nil, err
+	}
+
+	return balances.Sort(), res, nil
+}
+
```

Comment:
Let's move this inside of the `AllBalances` method. I see no reason this would be needed outside of gRPC, am I wrong? I thought I made this comment before...